codegen/templates/serverless-workflow.tmpl

```diff
@@ -0,0 +1,154 @@
+{{/* template to render gateway workflow interface code */ -}}
+{{- $instance := .Instance }}
+package workflow
+
+{{- $endpointType := .Spec.EndpointType }}
+{{- $reqHeaderMap := .ReqHeaders }}
+{{- $reqHeaderMapKeys := .ReqHeadersKeys }}
+{{- $reqHeaderRequiredKeys := .ReqRequiredHeadersKeys }}
+{{- $resHeaderMap := .ResHeaders }}
+{{- $resHeaderMapKeys := .ResHeadersKeys }}
+{{- $dummyEndpoint := .IsDummyEndpoint }}
+{{- $clientID := .ClientID }}
+{{- $clientName := title .ClientName }}
+{{- $clientMethodName := title .ClientMethodName }}
+{{- $serviceMethod := printf "%s%s" (title .Method.ThriftService) (title .Method.Name) }}
+{{- $workflowInterface := printf "%sDummyWorkflow" $serviceMethod }}
+{{- $workflowStruct := camel $workflowInterface }}
+{{- $method := .Method }}
+
+import (
+	"context"
+	"net/textproto"
+	"github.com/uber/zanzibar/config"
+
+	zanzibar "github.com/uber/zanzibar/runtime"
+
+	{{range $idx, $pkg := .IncludedPackages -}}
+	{{$pkg.AliasName}} "{{$pkg.PackageName}}"
+	{{end -}}
+
+	{{if .Method.Downstream }}
+	{{- range $idx, $pkg := .Method.Downstream.IncludedPackages -}}
+	{{$file := basePath $pkg.PackageName -}}
+	{{$pkg.AliasName}} "{{$pkg.PackageName}}"
+	{{end}}
+	{{- end}}
+
+	"go.uber.org/zap"
+	module "{{$instance.PackageInfo.ModulePackagePath}}"
+)
+
+{{with .Method -}}
+// {{$workflowInterface}} defines the interface for {{$serviceMethod}} workflow
+type {{$workflowInterface}} interface {
```

Comment:
This interface should not be redeclared, it should be common b/w various workflows. program to interface and not to implementation.